workflow/ttlcontroller/ttlcontroller.go

```diff
@@ -236,11 +239,19 @@ func timeLeft(wf *wfv1.Workflow, since *time.Time) (*time.Duration, *time.Time)
 	}
 }
 
-func getTTLStrategy(wf *wfv1.Workflow) *wfv1.TTLStrategy {
-	ttlStrategy := wf.Spec.TTLStrategy
-
-	if ttlStrategy == nil && wf.Status.StoredWorkflowSpec != nil && wf.Status.StoredWorkflowSpec.TTLStrategy != nil {
-		return wf.Status.StoredWorkflowSpec.TTLStrategy
+func getTTLStrategy(wf *wfv1.Workflow, defaultTTLStrategy *wfv1.TTLStrategy) *wfv1.TTLStrategy {
+	var ttlStrategy *wfv1.TTLStrategy
+	// TTLStrategy from Workflow default from Config
+	if defaultTTLStrategy != nil {
+		ttlStrategy = defaultTTLStrategy
+	}
+	// TTLStrategy from WorkflowTemplate
+	if wf.Status.StoredWorkflowSpec != nil && wf.Status.StoredWorkflowSpec.GetTTLStrategy() != nil {
```

Comment:
is this correct - that the store spec is low precedence than the original?

Reply:
Order of precedence:
1. Workflow
2. WorkflowTemplate
3. Workflowdefault